test/machineslib.py

```diff
@@ -247,6 +247,10 @@ def deleteDisk(self, target, vm_name="subVmTest1", expect_path=None):
 
         b.wait_visible(f"#vm-{vm_name}-disks-{target}-device")
         b.click(f"#vm-{vm_name}-disks-{target}-action-kebab")
+        b.wait_visible(f"#vm-{vm_name}-disks-{target}-action-kebab[aria-expanded=true]")
+        # HACK: kebab open state rendered in lists is bouncing
+        time.sleep(0.2)
+        b.wait_visible(f"#vm-{vm_name}-disks-{target}-action-kebab[aria-expanded=true]")
```

Comment:
So the theory is that the dropdown is closed and then reopened spontaneously, right?  I am not sure about opening/closing, but the `Dropdown` rabbit hole is very deep, with tens if not hundreds of `useEffect` and timers and at least two `useIsometricLayoutEffect`... It boggles the mind how complex it seems to be to get a dropdown menu on the screen. So I can believe that _something_ is going on behind the scenes that disturbs our test code somehow.

So this delay seems reasonable to me, to let the menu settle down after opening it. But we shouldn't give the impression that we know exactly what's going on, and maybe just say "HACK - PatternFly DropDown components seem to need some time to settle down after opening, sometimes, maybe."

Reply:
[This failure screenshot](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1539-1695d83f-20240405-095751-ubuntu-2204/TestMachinesStoragePools-testStoragePoolsBasic-ubuntu-2204-127.0.0.2-2301-FAIL.png) shows a open dropdown, but in a funny place. Something is going on there. Maybe the Kebab gets moved around, and maybe the Dropdown has to catch up? We could also try to insert a delay after `goToVmPage` to let the page itself settle down before opening the dropdown.

Reply:
> So the theory is that the dropdown is closed and then reopened spontaneously, right?

That's the observable and provable behaviour, yes. I originally thought it was the bouncing of the open-ness state due to delayed functional event handlers, but as you spotted I mis-read the output and this is actually fine. I also really believe that it is *not* related to missing `key=`s. First of all, React would complain about it, and second of all that's precisely why we keep a *global* open-ness state for all table entries, instead of tracking the state individually inside of the kebab.

So indeed I don't know at all what happens now. If the PF Dropdown is really so massively complicated (why oh why?), then maybe we should not use it at all, and implement our own simpler thing in cockpit-components-dropdown? Or go back to the deprecated one (but that's not a long-term option obviously). *throws hands into the air*.

With this kind of "we don't know what's going on", we should then introduce a testlib.py helper function for selecting something from a dropdown, and we can centrally add the debouncing there. I'm sure this affects other places, just perhaps less obviously due to sheer timing luck. WDYT?

Reply:
For the lolz, PF Dropdown also messes up in a completely different way: https://github.com/cockpit-project/starter-kit/pull/835

Reply:
> For the lolz, PF Dropdown also messes up in a completely different way: [cockpit-project/starter-kit#835](https://github.com/cockpit-project/starter-kit/pull/835)

Autsch...